### ISSUE TITLE
Person lookup fails when trying to add person to unit.

### ISSUE THREAD

**Issue by jhoerr** [indiana-university/itpeople-functions-v2#15]
Request: 

```
GET | https://itpeople-api-test.azurewebsites.net/people-lookup?q=vbhupat&_limit=15
```

Response:

```
{
  "StatusCode":500,
  "Errors": [ "Failed to search all people by netId" ],
  "Details":"When performing a set operation, both operands must have the same Include operations."
}
```

<img width="1358" alt="image" src="https://user-images.githubusercontent.com/1151206/111798943-edc22280-88a0-11eb-9658-989925e1b647.png">

**Comment by jasonfrancis** [comment]
This is one of those problems that turns out to be WAY harder than you'd expect in Entity Framework.

The immediate problem was using a constructor in the `.Select()` in `SearchHrPeopleByNameOrNetId()`.  Once you fix that you start getting even screwier errors when you try to `.Union()` or `.Concat()` the Person records from the People table with the Person records from the HrPeople table.

**Comment by jhoerr** [comment]
Ooof. That sounds gnarly.

**Comment by jasonfrancis** [comment]
I came up with a fix, and I can't decide if it's clever or terrible.
```c#
private static IQueryable<Person> SearchHrPeopleByNameOrNetId(PeopleContext db, HrPeopleSearchParameters query)
//To get an IQueryable<Person> that can be used in .Union()
// we need to psych-out EF by making it thinks it's pulling real Person records.
=> db.People
    .FromSqlInterpolated<Person>($"SELECT net_id, name, name_first, name_last, campus, position FROM hr_people")
    .Where(h=>  EF.Functions.ILike(h.Netid, $"%{query.Q}%")
        || EF.Functions.ILike(h.Name, $"%{query.Q}%"))
    .AsNoTracking();
```
Gets us an `IQueryable<Person>` that then works as expected in `SearchBothByNameOrNetId()`.

@jhoerr do you think it's worth doing this hack to get EF working, or should we just bring back the query from the old API?

**Comment by jhoerr** [comment]
I'm for whichever one is performant and feels more understandable/maintainable for y'all. It might be worth comparing the performance of the old query to this new solution. This is used by a typeahead, so performance is a factor.

**Comment by jasonfrancis** [comment]
I feel pretty good about the performance of fc289e2a especially once we started honoring the `_limit` query param.  I'll be interested to see how the performance stacks up against the real database.

I'll PR branch `issues/15` Monday.

**Comment by jasonfrancis** [comment]
Resolved with pull request #16
